src/parquet/column/reader.cc

```diff
@@ -176,12 +182,12 @@ bool TypedColumnReader<TYPE>::ReadNewPage() {
 // ----------------------------------------------------------------------
 // Batch read APIs
 
-static size_t DecodeMany(RleDecoder* decoder, int16_t* levels, size_t batch_size) {
+static size_t DecodeMany(LevelDecoder &decoder, int16_t* levels, size_t batch_size) {
```

Comment:
Can you push this code down into into `LevelDecoder`? The `Decode` API for LevelDecoder should be array oriented to be consistent with the other decoders in `parquet/encodings`